### PR TITLE
Add support for Cata Destruction Warlock 

### PR DESCRIPTION
**Pull request by emallson**
This builds on the [foundation guide PR](https://github.com/WoWAnalyzer/WoWAnalyzer/pull/6832) to add support for Destro Lock in Cata Classic. As part of this, I also added a couple of other missing bits like the popular Synapse Springs engineering item and updated the potion cooldown to match the actual cooldown in-game (1m, only starts if used pre-combat).

Lastly, since this involved adding a number of new spells I also added a "copy definition" helper to the debug view to streamline that process a little bit. Not as nice as something auto-importing it, but much easier than entering it all by hand.

This includes the changes from #6832. I recommend checking [this diff](https://github.com/WoWAnalyzer/WoWAnalyzer/compare/emallson:WoWAnalyzer:foundation-guide-1...emallson:WoWAnalyzer:foundation/cata-destro) for easier viewing. (I can't target this PR at that branch because doing so puts the PR on my fork instead of the main fork where people have proper permissions)

## Screenshots

image
image
image

### CODE REVIEW

File: src/analysis/classic/warlock/destruction/CHANGELOG.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { change, date } from 'common/changelog';
+import { emallson } from 'CONTRIBUTORS';
+
+export default [
+  // REMOVE THIS ENTRY AND ADD YOUR OWN
```
Comment (jazminite) [inline src/analysis/classic/warlock/destruction/CHANGELOG.tsx]: Remove comment

File: src/analysis/classic/warlock/destruction/modules/features/AlwaysBeCasting.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  suggestions(when: When) {
+    const boss = this.owner.boss;
+    if (!boss || !boss.fight.disableDowntimeSuggestion) {
+      when(this.downtimeSuggestionThresholds).addSuggestion((suggest, actual, recommended) =>
+        suggest(
+          <>
+            Your downtime can be improved. Try to Always Be Casting (ABC). If you have to move, use
+            instant cast spells, such as {/* UPDATE THE SPELLS BELOW */}
+            <SpellLink spell={SPELLS.FIRE_BLAST} /> or <SpellLink spell={SPELLS.ICE_LANCE} />.
```
Comment (jazminite) [inline src/analysis/classic/warlock/destruction/modules/features/AlwaysBeCasting.tsx]: Update these spells or remove the suggestion

File: src/analysis/classic/warlock/destruction/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// Reference this file in src\parser\index.ts
```
Comment (jazminite) [inline src/analysis/classic/warlock/destruction/index.ts]: Remove comment

**Review by jazminite — CHANGES_REQUESTED** [review]
Mostly looks good!

**Review by jazminite — APPROVED** [review]
Looks good!

**Comment by emallson** [comment]
This has been merged into the main foundation guide PR

**Pull request closed without merging**